client/components/forms/form-radio-with-thumbnail/index.jsx

```diff
@@ -0,0 +1,44 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import FormRadio from 'components/forms/form-radio';
+import FormLabel from 'components/forms/form-label';
+import cssSafeUrl from 'lib/css-safe-url';
+
+const FormRadioWithThumbnail = ( { label, thumbnail, ...otherProps } ) => {
+	const { cssClass, cssColor, imageUrl } = thumbnail;
+	const styles = {
+		backgroundColor: cssColor,
+		backgroundImage: imageUrl ? 'url(' + cssSafeUrl( imageUrl ) + ')' : '',
+	};
```

Comment:
> We want the component to be as flexible/reusable as possible.

In general yes, but it's important to not abstract early. Components should be flexible and reusable, but always tailored to specific, existing use cases, and be as clear and as simple as possible for that, until a new user case surfaces. Scoping for an hypothetical future should be avoided imho.

This as a general take. ;) 

> If we consider it being used in the context of user-generated images

We already have patterns for that (see media) and if a specific need arise, we'll design a new one that solves the issue, or adapt the above for that need. I can't design or even tell if this will work in a hypothetical scenario.

In terms of semantic, I think it would lead to more easily maintainable code for the scope as it would avoid splitting the point where the image is selected, to the point where it's declared. It's just simpler if the code specifies the image, without any indirection in either CSS nor a `style` rule.

Also, the above is similar in structure to the role that image+labels play: the selector is primary about the image (which is the content communicated) and the label is a description of the image. It's one of these cases where semantic precision ensures simpler code as well as accessible code.

As a minor, side benefit, we'd also avoid the tricky use of `style`, which would block any CSS override (to be clear: I'm not against using `style` directly, but needs to be done with care).